clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6294,6 +6294,75 @@ static void handleSYCLIntelPipeIOAttr(Sema &S, Decl *D,
   S.addSYCLIntelPipeIOAttr(D, Attr, E);
 }
 
+SYCLIntelFPGAMaxConcurrencyAttr *Sema::MergeSYCLIntelFPGAMaxConcurrencyAttr(
+    Decl *D, const SYCLIntelFPGAMaxConcurrencyAttr &A) {
+  // Check to see if there's a duplicate attribute with different values
+  // already applied to the declaration.
+  if (const auto *DeclAttr = D->getAttr<SYCLIntelFPGAMaxConcurrencyAttr>()) {
+    const auto *DeclExpr = dyn_cast<ConstantExpr>(DeclAttr->getNThreadsExpr());
+    const auto *MergeExpr = dyn_cast<ConstantExpr>(A.getNThreadsExpr());
+    if (DeclExpr && MergeExpr &&
+        DeclExpr->getResultAsAPSInt() != MergeExpr->getResultAsAPSInt()) {
+      Diag(DeclAttr->getLocation(),
+           diag::err_sycl_loop_or_function_attr_duplication)
+          << 1 << 1 << DeclAttr;
+      return nullptr;
+    }
+  }
+  // TODO
```

Comment:
I assume you're still working on this bit? :-)

Reply:
This TODO and the one below is a marker for the upcoming disable_component_pipelining attribute that is not implemented yet.

Reply:
If you're planning on doing that work in a follow-up rather than as part of this patch, can you switch this to a `FIXME` along with some comments about what's left to be fixed instead of `TODO` (community usually uses `FIXME` for areas where there's a known problem and `TODO` as a marker for work that's left to be done in a patch)?